src/main.rs

```diff
@@ -28,14 +99,47 @@ fn main() {
                 .value_name("UDF_FILE")
                 .help("Optionally sets .cc user defined function file to use"),
         )
+        .arg(
+             Arg::with_name("compilation_mode")
+                 .short("c")
+                 .long("compilation_mode")
+                 .value_name("COMPILATION_MODE")
+                 .takes_value(true)
+                 .possible_values(&compile_vals)
+                 .default_value("cpp")
+                 .help("Sets what to compile to:  the simulator (sim) or cpp wasm filter (cpp)"),
+         )
+        .arg(
+             Arg::with_name("root_node")
+                 .short("r")
+                 .long("root_node")
+                 .value_name("ROOT_NODE")
+                 .takes_value(true)
+                 .default_value("0")
+                 .help("Sets the root node of a query"),
+         )
+        .arg(
+             Arg::with_name("filter_name")
```

Comment:
This is likely redundant. The master branch already has an option for filter name.

Reply:
Rebased with master's code